File: MP.DataAccess/EntityModels/Movie.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace MP.DataAccess.EntityModels
{
    public class Movie
    {
        [Key]
        public int Id { get; set; }
        public string Title { get; set; }
        public string Language { get; set; }
        public string Image { get; set; }
        public string Overview { get; set; }
        public int Duration { get; set; }
        public IList<MoviesGenres> MoviesGenres { get; set; }
        public List<Genre> Genres { get; set; }
        public List<Show> Shows { get; set; }
    }
}

File: MP.Web/ClientApp/src/models/movie.ts
import IGenre from './genre';

enum MovieFields {
    TITLE = "title",
    LANGUAGE = "language",
    IMAGE = "image",
    OVERVIEW = "overview",
    DURATION = "duration",
    GENRES = "genres"
}

export default interface IMovie {
    id?: number,
    [MovieFields.TITLE]: string,
    [MovieFields.LANGUAGE]: string,
    [MovieFields.IMAGE]: string,
    [MovieFields.OVERVIEW]: string,
    [MovieFields.DURATION]: number,
    [MovieFields.GENRES]: IGenre[]
}